home/models.py

```diff
@@ -983,26 +984,42 @@ def update_embedding(sender, instance, *args, **kwargs):
     instance.embedding = embedding
 
 
-class OrderedContentSet(index.Indexed, models.Model):
+class OrderedContentSet(DraftStateMixin, RevisionMixin, index.Indexed, models.Model):
+    revisions = GenericRelation(
+        "wagtailcore.Revision", related_query_name="orderedcontentset"
+    )
     name = models.CharField(
         max_length=255, help_text="The name of the ordered content set."
     )
 
     def get_gender(self):
-        for item in self.profile_fields.raw_data:
+        for item in self.get_latest_revision_as_object().profile_fields.raw_data:
```

Comment:
Will this not mean that if we have changes in a new revision, then they will be returned in the API before that revision is published?

We only want published changes to be visible on the API, so maybe this is something we could add a unit test for? Create an ordered content set, publish it. Then make some changes to create a new revision, but don't publish that revision yet. Then check that the older published version appears on the API, but the newer changes only appear with the `qa` query param.

Reply:
I've added some more tests. For these methods in particular: I think they're only used by search (which I'm not entirely sure is working correctly), so they don't impact the API.